packages/api/src/services/lsp-state.ts

```diff
@@ -43,6 +44,28 @@ export default (config: Config, appState: Dependencies) => {
       ...result,
     };
   }
+
+  async function getStaticProps(instance: Instance, address: string) {
+    return batchRead(staticProps, instance, address);
+  }
+  async function getDynamicProps(instance: Instance, address: string) {
+    return batchRead(dynamicProps, instance, address);
+  }
+  // this is not ideal, but contract does not have a way to query this directly. this is a dyanmic value which
+  // depends first on gettin the static value 'collateralToken'.
+  async function getPositionCollateral(instance: Instance, address: string) {
+    const collateralToken = await instance.collateralToken();
+    return getErc20BalanceOf(collateralToken, address);
+  }
+  // get token state based on contract
+  async function getErc20BalanceOf(erc20Address: string, userAddress: string) {
```

Comment:
this does not consider the decimals of the collateral. is this something to add?

Reply:
good point let me look into this

Reply:
ok this is probably not ideal, but its not broken. currently the only place this is used is to calculate TVL. and that calculation does take into account collateral decimals when its calculated in stats. This might be confusing if someone wanted to read the value from the emp state but they will have access to collateralDecimals on the emp state when they query.

Maybe ideally everything gets stored in 18 dec but not sure if that change is worth it just yet